rust/src/basis.rs

```diff
@@ -217,12 +214,13 @@ impl Basis {
     /// bu.apply(1, TransposeMode::NoTranspose, EvalMode::Interp, &u, &mut v);
     ///
     /// // Verify results
-    /// let v_arr = v.view();
-    /// let x_qpts_arr = x_qpts.view();
-    /// for i in 0..Q {
-    ///   let true_value = x_qpts_arr[i]*x_qpts_arr[i]*x_qpts_arr[i] + 1.;
-    ///   assert_eq!(v_arr[i], true_value, "Incorrect basis application");
-    /// }
+    /// v.view()
+    ///     .iter()
+    ///     .zip(x_qpts.view().iter())
```

Comment:
We could impl [IntoIterator](https://doc.rust-lang.org/std/iter/trait.IntoIterator.html) for views so that an extra `iter()` isn't needed. Or probably implement `Iterator` for our vectors.

Reply:
Yeah, I'd like to impl iter for the vectors